js/web/lib/wasm/jsep/webgpu/ops/gather-elements.ts

```diff
@@ -0,0 +1,119 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import {DataType} from '../../../wasm-common';
+import {TensorView} from '../../tensor';
+import {ShapeUtil} from '../../util';
+import {AttributeWithCacheKey, createAttributeWithCacheKey} from '../attribute-with-cache-key';
+import {ComputeContext, GpuDataType, ProgramInfo, ProgramMetadata} from '../types';
+
+import {outputVariable, ShaderHelper} from './common';
+
+export interface GatherElementsAttributes extends AttributeWithCacheKey {
+  axis: number;
+}
+
+const validateInputs = (inputs: readonly TensorView[]): void => {
+  if (!inputs || inputs.length !== 2) {
+    throw new Error('GatherElements requires 2 inputs.');
+  }
+
+  if (inputs[0].dims.length < 1) {
+    throw new Error('GatherElements requires that the data input be rank >= 1.');
+  }
+
+  if (inputs[0].dims.length !== inputs[1].dims.length) {
+    throw new Error(`GatherElements requires that the data input and
+                     indices input tensors be of same rank.`);
+  }
+};
+
+const createGatherElementsProgramInfo =
+    (metadata: ProgramMetadata, inputs: readonly TensorView[], attributes: GatherElementsAttributes): ProgramInfo => {
+      const inputShape = inputs[0].dims;
+      const inputRank = inputShape.length;
+      const inputStrides = ShapeUtil.computeStrides(inputShape);
+
+      const indicesShape = inputs[1].dims;
+
+      const axis = ShapeUtil.normalizeAxis(attributes.axis, inputRank);
+      const axisDimLimit = inputShape[axis];
+
+      const outputShape = indicesShape.slice(0);
+      const outputSize = ShapeUtil.size(outputShape);
+
+      const inputDataType = inputs[0].dataType;
+      const elementSize = [DataType.int64, DataType.uint64, DataType.double].includes(inputDataType) ? 2 : 1;
+      const indicesElementSize = inputs[1].dataType === DataType.int64 ? 2 : 1;
```

Comment:
if use IndicesHelper to deal with inputIndices it may be simpler - it can handle int64/int32 and their value type are both i32

Reply:
Not sure if I am mis-understanding your comment but `inputIndices` is not really an N-D indices - it is the value of indices along the `axis` attribute to gather data from the input.

Reply:
Updated shader based on offline sync